consumer/subscriber_data_management.go

```diff
@@ -55,68 +86,108 @@ func SDMGetAmData(ue *amf_context.AmfUe) (problemDetails *models.ProblemDetails,
 	} else if httpResp != nil {
 		if httpResp.Status != localErr.Error() {
 			err = localErr
-			return
+			return nil, err
 		}
 		problem := localErr.(openapi.GenericOpenAPIError).Model().(models.ProblemDetails)
 		problemDetails = &problem
 	} else {
 		err = openapi.ReportError("server no response")
 	}
-	return
+	return problemDetails, err
 }
 
-func SDMGetSmfSelectData(ue *amf_context.AmfUe) (problemDetails *models.ProblemDetails, err error) {
+func SDMGetSmfSelectData(ctx context.Context, ue *amf_context.AmfUe) (*models.ProblemDetails, error) {
+	ctx, span := tracer.Start(ctx, "HTTP GET udm/{supi}/smf-select-data")
+	defer span.End()
+
+	span.SetAttributes(
+		attribute.String("http.method", "GET"),
+		attribute.String("nf.target", "udm"),
+		attribute.String("net.peer.name", ue.NudmSDMUri),
+		attribute.String("udm.supi", ue.Supi),
+		attribute.String("plmn.id", ue.PlmnId.Mcc+ue.PlmnId.Mnc),
+	)
+
 	configuration := Nudm_SubscriberDataManagement.NewConfiguration()
 	configuration.SetBasePath(ue.NudmSDMUri)
 	client := Nudm_SubscriberDataManagement.NewAPIClient(configuration)
 
 	paramOpt := Nudm_SubscriberDataManagement.GetSmfSelectDataParamOpts{
 		PlmnId: optional.NewInterface(ue.PlmnId.Mcc + ue.PlmnId.Mnc),
 	}
-	ctx, cancel := context.WithTimeout(context.TODO(), 30*time.Second)
+	ctx, cancel := context.WithTimeout(ctx, 30*time.Second)
 	defer cancel()
+
+	var err error
+	var problemDetails *models.ProblemDetails
+
```

Comment:
Same as my previous comments
```suggestion
func SDMGetSmfSelectData(ctx context.Context, ue *amf_context.AmfUe) (problemDetails *models.ProblemDetails, err error) {
	ctx, span := tracer.Start(ctx, "HTTP GET udm/{supi}/smf-select-data")
	defer span.End()

	span.SetAttributes(
		attribute.String("http.method", "GET"),
		attribute.String("nf.target", "udm"),
		attribute.String("net.peer.name", ue.NudmSDMUri),
		attribute.String("udm.supi", ue.Supi),
		attribute.String("plmn.id", ue.PlmnId.Mcc+ue.PlmnId.Mnc),
	)

	configuration := Nudm_SubscriberDataManagement.NewConfiguration()
	configuration.SetBasePath(ue.NudmSDMUri)
	client := Nudm_SubscriberDataManagement.NewAPIClient(configuration)

	paramOpt := Nudm_SubscriberDataManagement.GetSmfSelectDataParamOpts{
		PlmnId: optional.NewInterface(ue.PlmnId.Mcc + ue.PlmnId.Mnc),
	}
	ctx, cancel := context.WithTimeout(ctx, 30*time.Second)
	defer cancel()

```

Reply:
Done